src/main/java/gicha/service/CharacterService.java

```diff
@@ -0,0 +1,184 @@
+package gicha.service;
+
+import gicha.Character;
+import gicha.exceptions.CharacterNotFoundException;
+import gicha.model.character.Birthplace;
+import gicha.model.character.CharacterDtoToCharacter;
+import gicha.model.character.Element;
+import gicha.repository.CharacterRepository;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+/**
+ * Service class for character controller.
+ *
+ * @author Lavrek
+ */
+@Service
+@AllArgsConstructor
+public class CharacterService {
+    CharacterRepository characterRepository;
+    CharacterDtoToCharacter characterDtoToCharacter;
+
+    /**
+     * //     * Save a character in the database
+     * //     * The url argument must contain Character object
+     * //     *
+     * //     * @param character Character object
+     * //
+     */
+    public void save(Character character) {
+        characterRepository.save(characterDtoToCharacter.characterToCharacterDto(character));
+    }
+
+    /**
+     * Returns a Character object
+     * The url argument must specify character ID
+     *
+     * @param id the character ID
+     * @return The character at the specified URL
+     * @throws CharacterNotFoundException if there is no character with specified ID
+     * @see Character
+     */
+    public Character findById(Long id) {
+        return characterRepository.findById(id).map(e -> characterDtoToCharacter.characterDtoToCharacter(e))
+                .orElseThrow(CharacterNotFoundException::new);
+    }
+
+    /**
+     * Delete a character from database
+     * The url argument must specify character ID
+     *
+     * @param id the character ID
+     * @throws CharacterNotFoundException if there is no characters with specified ID
+     */
+    public void deleteById(Long id) {
+        gicha.model.character.Character characterToDelete = characterRepository.findById(id).orElseThrow(CharacterNotFoundException::new);
+        characterRepository.deleteById(characterToDelete.getId());
+    }
+
+    /**
+     * Returns all characters from database
+     *
+     * @return List of characters
+     * @throws CharacterNotFoundException if there aren't any character in database
+     * @see List
+     */
+    public List<Character> getAllCharacters() {
+        List<Character> characterList = characterRepository.findAll().stream()
+                .map(e -> characterDtoToCharacter.characterDtoToCharacter(e))
+                .toList();
+        if (characterList.size() > 0) {
```

Comment:
Collections.isEmpty()

Reply:
da se prepsat primo ten stream btw aby rovnou vracel list nebo exception, ale nevadi to.